### PR TITLE
Smoke test for Checkout service

### PR DESCRIPTION
**Pull request by amork**
Grant execute and describe actions to the policy statement
Implement elements of the smoke test for 
- initiating the checkout 
- checking status of the checkout job

Ran smoke test.

connect to #639

### CODE REVIEW

File: tests/test_smoketest.py [hunk 1]
```diff
@@ -111,6 +111,22 @@ def test_smoketest(self):
             run(f"{venv_bin}hca dss get-subscriptions --replica {replica}")
             run(f"{venv_bin}hca dss delete-subscription --replica {replica} --uuid {sub_id}")
 
+        run(f"{venv_bin}hca dss post-bundles-checkout "
+            "--uuid 00251ba0-39fc-4afe-9249-891a3de0f5ba "
```
Comment (kislyuk) [inline tests/test_smoketest.py]: Why is there a hardcoded UUID here? I think this should be the uuid of the bundle that was uploaded by smoketest earlier (`--uuid $(jq -r .bundle_uuid upload.json)`).

File: tests/test_smoketest.py [hunk 2]
```diff
@@ -111,6 +111,22 @@ def test_smoketest(self):
...
+            "--replica aws "
+            "--email noreply@humancellatlas.org >res.json")
```
Comment (kislyuk) [inline tests/test_smoketest.py]: Please add a space after `>`.

File: tests/test_smoketest.py [hunk 3]
```diff
@@ -111,6 +111,22 @@ def test_smoketest(self):
...
+        with open("res.json") as fh:
+            res = json.load(fh)
+            print(f"Checkout jobId: {res['checkout_job_id']}")
+            assert len(res["checkout_job_id"]) > 0
+
+        run(f"{venv_bin}hca dss get-bundles-checkout --checkout-job-id {res['checkout_job_id']} >res.json")
```
Comment (kislyuk) [inline tests/test_smoketest.py]: Please add a space after `>`.

File: tests/test_smoketest.py [hunk 4]
```diff
@@ -111,6 +111,22 @@ def test_smoketest(self):
...
+        with open("res.json") as fh:
+            res = json.load(fh)
+            print(f"Checkout jobId: {res['status']}")
+            #TODO (rkisin) once email sender has been authorized, change this assert to a waiter for "SUCCESS" status
```
Comment (kislyuk) [inline tests/test_smoketest.py]: Please fix the lint error here.

File: tests/test_smoketest.py (hunk 1 above)
Comment (amork) [reply tests/test_smoketest.py]: fixed

File: tests/test_smoketest.py (hunk 2 above)
Comment (amork) [reply tests/test_smoketest.py]: fixed

File: tests/test_smoketest.py (hunk 3 above)
Comment (amork) [reply tests/test_smoketest.py]: fixed

**Comment by kislyuk** [comment]
Hi @amork, can you please take a look at the Travis CI build? You may need rebasing to make the build green.

**Comment by kislyuk** [comment]
How long does the checkout job take? Ideally we should kick it off immediately after bundle creation, and wait for it to complete at the end of the smoketest.

**Comment by amork** [comment]
@kislyuk kick off checkout immediately after bundle is downloaded.  Waiting for completion at the end of the smoketest with up to 60 seconds timeout.

@ttung added check for number of files in the test bundle. Parsing json upon the bundle manifest download to get number of files. Upon successful checkout get file count from the target location and comparing the file count vs expected count from json.

File: tests/test_smoketest.py [hunk 5]
```diff
@@ -83,6 +88,18 @@ def test_smoketest(self):
             "--staging-bucket $DSS_S3_BUCKET_TEST "
             f"--src-dir {bundle_dir} > upload.json")
         run(f"{venv_bin}hca dss download --replica aws --bundle-uuid $(jq -r .bundle_uuid upload.json)")
+
+        file_count = int(get_upload_val(".files | length"))
+
+        run(f"{venv_bin}hca dss post-bundles-checkout "
+            "--uuid $(jq -r .bundle_uuid upload.json) "
+            "--replica aws "
+            "--email rkisin@chanzuckerberg.com > res.json")
```
Comment (kislyuk) [inline tests/test_smoketest.py]: Probably best to set this to dss.humancellatlas@gmail as well

File: tests/test_smoketest.py [hunk 6]
```diff
@@ -115,6 +141,28 @@ def test_smoketest(self):
...
+        bundle_id = get_upload_val(".bundle_uuid")
+        version = get_upload_val(".version")
+
+        for i in range(10):
+            run(f"{venv_bin}hca dss get-bundles-checkout "
+                f"--checkout-job-id {res_checkout['checkout_job_id']} > res.json")
+            with open("res.json") as fh:
+                res = json.load(fh)
+                status = res['status']
+                assert len(status) > 0
```
Comment (kislyuk) [inline tests/test_smoketest.py]: self.assertGreater

File: tests/test_smoketest.py [hunk 7]
```diff
@@ -115,6 +141,28 @@ def test_smoketest(self):
...
+                if status == 'RUNNING':
+                    time.sleep(6)
+                else:
+                    assert status == 'SUCCEEDED'
```
Comment (kislyuk) [inline tests/test_smoketest.py]: self.assertEqual

File: tests/test_smoketest.py [hunk 8]
```diff
@@ -115,6 +141,28 @@ def test_smoketest(self):
...
+                    blob_handle = S3BlobStore()
+                    object_key = get_dst_bundle_prefix(bundle_id, version)
+                    print(f"Checking bucket {checkout_bucket} object key: {object_key}")
+                    files = blob_handle.list(checkout_bucket, object_key)
+                    self.assertEqual(sum(1 for _ in files), file_count)
```
Comment (kislyuk) [inline tests/test_smoketest.py]: len(files)?

File: tests/test_smoketest.py [hunk 9]
```diff
@@ -115,6 +141,28 @@ def test_smoketest(self):
             run(f"{venv_bin}hca dss get-subscriptions --replica {replica}")
             run(f"{venv_bin}hca dss delete-subscription --replica {replica} --uuid {sub_id}")
 
+        checkout_bucket = os.environ["DSS_S3_CHECKOUT_BUCKET"]
+        bundle_id = get_upload_val(".bundle_uuid")
```
Comment (kislyuk) [inline tests/test_smoketest.py]: If you're going to load this into a variable, then please hoist it to immediately after bundle creation, and replace the shell subprocess calls (`$(jq -r .bundle_uuid upload.json)`) with format string substitutions (`{bundle_id}`).

File: tests/test_smoketest.py (hunk 6 above)
Comment (amork) [reply tests/test_smoketest.py]: fixed

File: tests/test_smoketest.py (hunk 7 above)
Comment (amork) [reply tests/test_smoketest.py]: fixed

File: tests/test_smoketest.py (hunk 8 above)
Comment (amork) [reply tests/test_smoketest.py]: fixed

**Pull request merged**